### DISCUSSION TITLE
Does `term` work with `arrays`?

### DISCUSSION THREAD

**Discussion by deepjyoti30** [zinclabs/zinc#630] [Q&A]
I have a mapping where the type is `array` and I have a query that needs to make a term match for a value in the array. ElasticSearch supports using a `term` query for matching the value in array, however, when I try the same with Zinc it doesn't work.

Here's an example doc:

```JSON
{
   "tags": ["zinc", "elasticsearch"],
   ...
}
```

My query is:

```JSON
"query": {
  "bool": {
     "must": [
         {
            "term": {
                "tags": "elasticsearch"
             }
         }
      ]
   }
}
```

The above query doesn't return any results thought it should match the document that has the `elasticsearch` string in it.

However, `match` works in the above query instead of `term`:

```JSON
"query": {
  "bool": {
     "must": [
         {
            "match": {
                "tags": "elasticsearch"
             }
         }
      ]
   }
}
```

My question is, is this a bug or is this intentionally not supported yet? If it's not supported yet, do you guys have plans of supporting it in the future?

**Comment by hengfeiyang** [comment]
<img width="1184" alt="image" src="https://user-images.githubusercontent.com/1628250/199479532-37e9519f-ce4a-4926-8330-0074d0f2085b.png">
In my test it can work. Can you give some detail about your ingest data. and the query. and the version of zinc.
